httpe/err.go

```diff
@@ -0,0 +1,59 @@
+package httpe
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+)
+
+var (
+	ErrInvalidHeader    = fmt.Errorf("invalid header")
+	ErrMalformedJSON    = fmt.Errorf("malformed JSON")
+	ErrMethodNotAllowed = &ErrStatus{
+		Err:        fmt.Errorf("method not allowed"),
+		HTTPStatus: http.StatusMethodNotAllowed,
+	}
+)
+
+type ErrStatus struct {
+	Err        error
+	HTTPStatus int
+}
+
+func Err(err error, httpStatus int) error {
+	return &ErrStatus{Err: err, HTTPStatus: httpStatus}
+}
+
+func errStatus(err error) *ErrStatus {
+	var e *ErrStatus
+	if ok := errors.As(err, &e); ok {
+		return e
+	}
+	return &ErrStatus{Err: err, HTTPStatus: http.StatusInternalServerError}
+}
+
+func (e *ErrStatus) Error() string {
+	return e.Err.Error()
+}
+
+func (e *ErrStatus) HTTPMessage() string {
+	if e.Err == nil || e.HTTPStatus == 0 || e.HTTPStatus == http.StatusInternalServerError {
+		return "internal server error"
+	}
+	return e.Error()
+}
+
+func WriteErrTxt(w http.ResponseWriter, err error) {
+	e := errStatus(err)
+	http.Error(w, e.HTTPMessage(), e.HTTPStatus)
+}
+
+func WriteErrJSON(w http.ResponseWriter, err error) {
+	e := errStatus(err)
+	b, _ := json.Marshal(map[string]string{
+		"errorMessage": e.HTTPMessage(),
```

Comment:
nit: perhaps just "message"? this is an error response.

Reply:
yes, agreed. Thank you.

Reply:
hmm, actually - not sure... this is just a placeholder anyway.
the point is:
you get back some kind of json and it'd be nice to tell at a glance if it's an error...
all the api connect messages we've seen in another place also have `errorMessage`, `errorCode` etc. iirc. I am fine with that personally.

Reply:
I did consider that and it is a good reason to keep it. I erred on the side of brevity as the http status code indicates this is an error. If you got this on a 200 or 4/500, the "error" prefix makes sense. But if the http status is an error status then perhaps it is redundant.

Anyway, I'm nitpicking. It's your code, you do it as you prefer.